modules/server/src/network/resolvers/aggregations.ts

```diff
@@ -1,101 +1,120 @@
 		return;
 	}
 };
 
-/**
- * TODO: Placeholder for resolving multiple remote connection responses into single aggregation
- */
-const resolveAggregation = (response: any, type: string) => {};
+export const queryConnections = async (queries: NetworkQuery[]) => {
+	const networkQueryPromises = queries.map(async (query) => {
+		const response = await queryConnection(query);
+		return { config, response };
+	});
 
-/**
- * Creates resolver function
- * @param field
- * @returns resolver function
- */
-const createResolver = (field: SupportedNetworkFieldType, configs: NetworkAggregationConfig[]) => {
-	const connectionsToQuery = getConnectionURLs(field, configs);
-
-	return async () => {
-		const connectionResponses = await Promise.allSettled(
-			connectionsToQuery.map((url: string) => queryRemoteConnection(url, field.type)),
-		);
-
-		const responses = connectionResponses.filter(
-			(result: { status: string }) => result.status === 'fulfilled',
-		);
+	// TODO: expand on network condition handling, eg. timeouts, single connection failure
+	const networkResults = await Promise.allSettled(networkQueryPromises);
+	return networkResults;
+};
 
-		const resolvedData = resolveAggregation(responses, field.type);
+type NetworkQuery = {
+	url: string;
+	gqlQuery: DocumentNode;
+};
 
-		/**
-		 * TODO: return resolvedData
-		 */
-		return { test: field.name };
-	};
+/**
+ * Find requested field in remote connection supported fields
+ *
+ * @param config
+ * @param fieldName
+ * @returns
+ */
+const findMatchedAggregationField = (config: NetworkAggregationConfig, fieldName: string) => {
+	return config.supportedAggregations.find((agg) => agg.name === fieldName);
 };
 
 /**
- * Returns a resolver map of functions
+ * Parse central query and build individual queries for remote connections based on available fields
+ *
  * @param configs
- * @param allTypeDefs
- * @returns resolver map
+ * @param info
+ * @returns
  */
-export const createAggregationResolvers = (
+export const createNetworkQueries = (
 	configs: NetworkAggregationConfig[],
-	networkFieldTypes: SupportedNetworkFieldType[],
-) => {
-	return networkFieldTypes.reduce(
-		(resolvers, field) => ({ ...resolvers, [field.name]: createResolver(field, configs) }),
-		{},
-	);
+	info: GraphQLResolveInfo,
+): NetworkQuery[] => {
+	const rootQueryFields = getRootFields(info);
+
+	const queries = configs
+		.map((config) => {
+			// construct gql string { [fieldName] { [Aggregation query fields] } }
+			const gqlString = rootQueryFields.reduce((gqlString, fieldName) => {
+				const matchedAggregationField = findMatchedAggregationField(config, fieldName);
+				if (matchedAggregationField) {
+					const { name, type } = matchedAggregationField;
+					// get gql query string for supported aggregation
+					const aggregationFieldQueryString = supportedAggregationQueries.get(type);
+					return gqlString + `${name}${aggregationFieldQueryString}`;
+				}
+				return gqlString;
+			}, '');
```

Comment:
Will we get an invalid gql query if there are no matching aggregationFields for that node? How is this empty query handled?